src/lib.rs

```diff
@@ -40,24 +54,62 @@ impl TsnSocket {
     }
 }
 
-const CONTROL_SOCK_PATH: &str = "/var/run/tsn.sock";
-
-fn send_cmd(command: String) -> Result<String, std::io::Error> {
-    let mut stream = UnixStream::connect(CONTROL_SOCK_PATH)?;
-    stream.write_all(command.as_bytes())?;
-    let mut msg = String::new();
-    stream.read_to_string(&mut msg)?;
-    Ok(msg)
-}
-
-fn create_vlan(ifname: &str, vlanid: u16) -> Result<String, std::io::Error> {
-    let command = format!("create {} {}\n", ifname, vlanid);
-    send_cmd(command)
+fn create_vlan(ifname: &str, vlanid: u16) -> Result<String, String> {
+    let config = get_config(ifname)?;
+    let shm_name = get_shmem_name(ifname, vlanid);
+    let shm_fd = get_shmem_fd(&shm_name)?;
+    lock_shmem(&shm_fd)?;
+    let mut vlan_vec = read_shmem(&shm_name)?;
+    let name = if ifname.len() > 11 {
+        format!("{}.{}", &ifname[..10], vlanid)
+    } else {
+        format!("{}.{}", &ifname, vlanid)
+    };
```

Comment:
Split it as a function and it seems weird.
```
let name = if ifname.len() > 11 {
        format!("{}.{}", &ifname[..10], vlanid)
```
means if ifname length is more than 11, cut it to 10
Usually, if length is more than X, cut it to X

Reply:
`
let mut name = format!("{}.{}", ifname, vlanid);

if name.length > 15 {
       name = format!("{}.{}", &ifname[..10], vlaind);
}`
is better than before?